event-sauce/src/event_builder/conflict_event.rs

```diff
@@ -0,0 +1,80 @@
+//! Event builder
+
+use crate::event_builder::EventBuilder;
+use crate::{ConflictData, Entity, Event, EventData};
+use chrono::Utc;
+use uuid::Uuid;
+
+/// Event conflict builder
+pub struct ConflictEventBuilder<EDA: EventData, EDC: EventData> {
+    payload: ConflictData<EDA, EDC>,
+    session_id: Option<Uuid>,
+}
+
+impl<EDA, EDC> ConflictEventBuilder<EDA, EDC>
+where
+    EDA: EventData,
+    EDC: EventData,
+{
+    /// Consume the builder and produce the final event
+    pub fn build(self, entity: &EDA::Entity) -> Event<ConflictData<EDA, EDC>> {
+        Event {
+            id: Uuid::new_v4(),
+            event_type: String::from(self.payload.event_type()),
+            entity_type: EDA::Entity::entity_type(),
+            entity_id: entity.entity_id(),
+            session_id: self.session_id,
+            purger_id: None,
+            created_at: Utc::now(),
+            purged_at: None,
+            data: Some(self.payload),
+        }
+    }
+
+    /// Workaround method to get an entity ID out of entities when implementing
+    /// `ConflictEntityBuilder`
+    pub(crate) fn build_with_entity_id(self, entity_id: Uuid) -> Event<ConflictData<EDA, EDC>> {
+        Event {
+            id: Uuid::new_v4(),
+            event_type: String::from(self.payload.event_type()),
+            entity_type: EDA::Entity::entity_type(),
+            entity_id,
```

Comment:
I think the entity ID can be reused from `self.payload`, no? Then the `entity_id` argument to this method can be removed.

```suggestion
            self.payload.applied_event.entity_id,
```